WiinUPro/Controls/NintyControl.xaml.cs

```diff
@@ -394,6 +394,32 @@ public void LoadProfile(string fileName)
 
             if (loadedProfile != null)
             {
+                if (fileName.Length > 6 & fileName[fileName.Length - 6] == '_')
+                {
+                    int n = (int)Char.GetNumericValue(fileName[fileName.Length - 5]);
+                    bool changeProfile = false;
+                    if (n == 1 && ScpDirector.Access.IsConnected(ScpDirector.XInput_Device.Device_A))
+                    {
+                        changeProfile = true;
+                    }
+                    else if (n == 2 && ScpDirector.Access.IsConnected(ScpDirector.XInput_Device.Device_B))
+                    {
+                        changeProfile = true;
+                    }
+                    else if (n == 3 && ScpDirector.Access.IsConnected(ScpDirector.XInput_Device.Device_C))
+                    {
+                        changeProfile = true;
+                    }
+                    else if (n == 4 && ScpDirector.Access.IsConnected(ScpDirector.XInput_Device.Device_D))
+                    {
+                        changeProfile = true;
+                    }
+                    if (changeProfile && !_rumbleSubscriptions[n - 1])
+                    {
+                        LoadProfile(fileName.Substring(0, fileName.Length - 5) + (char)(49 + n) + ".wup");

```

Comment:
(char)48 = 0, (char)49 = 1 etc.

Reply:
Cleaver, but easier to read would be taking that last split part mentioned above then using a Replace with the part string  and then a string.Format or what's called an interpolated string `$"_{n}.wup"`